utils/types/servermode/servermode.go

```diff
@@ -0,0 +1,29 @@
+package servermode
+
+type Mode string
+
+const (
+	UndefinedMode Mode = ""
```

Comment:
One way to get away with `UndefinedMode` which doesn't look that much different from `DegradedMode` so far, would be to use int enum types, e.g.

```go
type Mode int

const (
	Degraded Mode = iota
        Normal
)

func (m Mode) String() string {
	return [2]string{"NORMAL", "DEGRADED"}[m]
}

func ValueOf(m string) (Mode, error) {

	switch m {
	case "NORMAL":
		return Normal, nil
	case "DEGRADED":
		return Degraded, nil
	}
	return Degraded, fmt.Errorf("unknown mode %s", m)

}
```

Reply:
We have removed the undefined mode actually. 
We need a parse/validator method, not sure if we need iotas seems a bit overkill.

Reply:
Thinking more about it, I like the `iota` approach, I can see the benefits.

Reply:
I was thinking about writing a generator for sparing us with the boilerplate code, but looks like someone else caught me already!
https://github.com/abice/go-enum

As a bonus, this generator handles json marshalling/unmarshalling concerns too, which allows us to se enum types in structs used for json binding